tests/python/sdk/test_attributes.py

```diff
@@ -0,0 +1,102 @@
+# Copyright (C) CVAT.ai Corporation
+#
+# SPDX-License-Identifier: MIT
+
+import pytest
+from cvat_sdk import models
+from cvat_sdk.attributes import attribute_value_validator, number_attribute_values
+
+
+def test_number_attribute_values_good():
+    assert number_attribute_values(0, 0, 1) == ["0", "0", "1"]
+    assert number_attribute_values(0, 10, 1) == ["0", "10", "1"]
+    assert number_attribute_values(0, 10, 10) == ["0", "10", "10"]
+    assert number_attribute_values(0, 10, 5) == ["0", "10", "5"]
+
+
+def test_number_attribute_values_bad():
+    with pytest.raises(ValueError, match="min_value must be less than or equal to max_value"):
+        number_attribute_values(1, 0, 1)
+
+    with pytest.raises(ValueError, match="step must be positive"):
+        number_attribute_values(0, 10, 0)
+
+    with pytest.raises(ValueError, match="step must be positive"):
+        number_attribute_values(0, 10, -1)
+
+    with pytest.raises(ValueError, match="step must be a divisor of max_value - min_value"):
+        number_attribute_values(0, 10, 3)
+
+
+def test_attribute_value_validator_checkbox():
+    validator = attribute_value_validator(
+        models.AttributeRequest(name="a", mutable=False, input_type="checkbox", values=[])
+    )
+
+    assert validator("true")
+    assert validator("false")
+    assert not validator("maybe")
+
+
+def test_attribute_value_validator_number():
+    validator = attribute_value_validator(
+        models.AttributeRequest(
+            name="a", mutable=False, input_type="number", values=["0", "10", "2"]
+        )
+    )
+
+    assert validator("0")
+    assert validator("2")
+    assert validator("10")
+    assert not validator("1")
+    assert not validator("-2")
+    assert not validator("12")
+    assert not validator("not a number")
+
+
+@pytest.mark.parametrize(
+    ["values", "exc_match"],
+    [
+        (["0", "1"], "wrong number of values"),
+        (["0", "10", "1", "1"], "wrong number of values"),
+        # the exception is raised by external code, so we don't want to rely on the message
+        (["a", "10", "1"], ""),
+        (["0", "a", "1"], ""),
+        (["0", "10", "a"], ""),
```

Comment:
Shouldn't these cases be wrapped by the sdk code?

Reply:
I don't know. Why bother?

Reply:
It's an error message that is supposed to be understood by the user, so it's a part of the library api.

Reply:
Then it should probably be checked in these tests

Reply:
But we can't be sure that the message won't be changed in a future Python version.

Reply:
I'm confident that the message will remain understandable, I just can't be sure of the exact words. 🤷 

Anyway, I don't want to die on this hill, so I went ahead and added a custom message.